wlauto/workloads/uxperf_applaunch/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,226 @@
+import java.util.Map;
+import java.util.*;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Map.Entry;
+import java.lang.*;
+import dalvik.system.DexClassLoader;
+import java.lang.reflect.Method;
+import java.net.*;
+
+
+public class UiAutomation extends BaseUiAutomation {
+
+
+	//Uiobject that marks the end of launch of an application.
+	//This is workload spefific and added in the workload Java file
+	//by a method called getLaunchEndObject().
+	public UiObject launchEndObject;
+	//Timeout to wait for application launch to finish. 
+	private Integer launch_timeout = 10;
+	public String applaunchType;
+	public String applaunchIterations;
+	public String activityName;
+	public ApplaunchInterface launch_workload;
+
+	//uiautomator function called by the uxperfapplaunch workload.
+	public void runUiAutomation() throws Exception{
+		parameters = getParams();
+
+		//Get workload jar file parameters
+		String workload = parameters.getString("workload");
+		String workloadJarPath = parameters.getString("workloadJarPath");
+		String workloadJarName = String.format("com.arm.wlauto.uiauto.%1s.jar",workload);
+		String workloadJarFile = String.format("%1s/%2s",workloadJarPath, workloadJarName);
+
+        //Load the jar file
+		File jarFile = new File(workloadJarFile);
+		if(!jarFile.exists()){
+			Log.e("Jar file not found:", workloadJarFile);
+		}
+		
+		DexClassLoader classloader = new DexClassLoader(jarFile.toURI().toURL().toString(), workloadJarPath, null, ClassLoader.getSystemClassLoader());
+
+		Class uiautomation = null;
+		Object uiautomation_interface = null;
+		try {
+			String workloadClass = String.format("com.arm.wlauto.uiauto.%1s.UiAutomation",workload);
+			uiautomation = classloader.loadClass(workloadClass);
+			Log.d("Class loaded:", uiautomation.getCanonicalName());
+			uiautomation_interface = uiautomation.newInstance();
+		} catch (ClassNotFoundException e) {
+			e.printStackTrace();
+		}
+		//Create an Application Interface object from the workload
+		launch_workload = ((ApplaunchInterface)uiautomation_interface);
+
+		//Get parameters for application launch
+		getPackageParameters();
+        applaunchType = parameters.getString("applaunch_type");
+        applaunchIterations = parameters.getString("applaunch_iterations");
+        activityName = parameters.getString("launch_activity");
+
+		//Run the workload for application launch initialization
+		runApplaunchSetup();
+		//Run the workload for application launch measurement
+		for (int iteration = 0; iteration < Integer.parseInt(applaunchIterations); iteration++) {
+        	Log.d("Applaunch iteration number: ", applaunchIterations);
+			sleep(20);
+			killBackground();
+			runApplaunchIteration(iteration);
+			closeApplication();
+		}
+	}
+    
+
+    //Setup run for uxperfapplaunch workload that clears the initial
+	//run dialogues on launching an application package.
+    public void runApplaunchSetup() throws Exception{
+        sleep(5);
```

Comment:
removed the delay, it was needed for previous implementation while I tested for many devices and workloads. I can do this later if I still have failures.